ui/conftest.py

```diff
@@ -15,44 +16,46 @@
     "devnet": "NeonEVM DevNet",
 }
 
+CHROME_TAR_PATH = pathlib.Path(__file__).absolute().parent / "extensions" / "data"
+CHROME_DATA_PATH = pathlib.Path(__file__).absolute().parent.parent / "chrome-data" / uuid.uuid4().hex
+"""CHROME_DATA_PATH is temporary local destination in project to untar chrome data directory and plugins"""
 
-CHROME_EXT_DIR = "extensions/chrome/plugins"
-"""Relative path to Chrome extension source
-"""
 
-CHROME_USER_DATA_DIR = "user_data"
-"""Relative path to Chrome extensions user data
-"""
+@pytest.fixture(scope="session")
+def network(pytestconfig: tp.Any) -> tp.Optional[str]:
+    return EVM_NETWORKS.get(pytestconfig.getoption("--network"), EVM_NETWORKS["devnet"])
 
 
-def pytest_addoption(parser):
-    parser.addoption("--network", action="store", default="devnet", help="Which stand use")
+@pytest.fixture(scope="session")
+def neonpass_url(pytestconfig: tp.Any) -> tp.Optional[str]:
+    return pytestconfig.environment.neonpass_url
 
 
 @pytest.fixture(scope="session")
-def network(pytestconfig: tp.Any) -> tp.Optional[str]:
-    return EVM_NETWORKS.get(pytestconfig.getoption("--network"), EVM_NETWORKS["devnet"])
+def solana_url(pytestconfig: tp.Any) -> tp.Optional[str]:
+    return pytestconfig.environment.solana_url
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def chrome_extensions_path(required_extensions: tp.Union[tp.List, str]) -> pathlib.Path:
-    path = ""
+    """Extracting Chrome Plugins"""
+    result_path = ""
     if isinstance(required_extensions, str):
         required_extensions = [required_extensions]
     for ext in required_extensions:
-        source = pathlib.Path(__file__).parent / CHROME_EXT_DIR / ext
-        if not path:
-            path = source
+        source = libs.extract_tar_gz(CHROME_TAR_PATH / f"{ext}.extension.tar.gz", CHROME_DATA_PATH / "plugins") / ext
+        if not result_path:
+            result_path = source
         else:
-            path = path / f",{source}"
-    return path
+            result_path = result_path / f",{source}"
+    yield result_path
+    libs.rm_tree(CHROME_DATA_PATH)
 
 
-@pytest.fixture(scope="session", autouse=True)
+@pytest.fixture(scope="function", autouse=True)
```

Comment:
why you change this to function? It's slower and we don't need to unpack data for every test

Reply:
This is made on purpose — both Phantom and MM plugins require wallet unlock step, but if we don't clean `chrome_data` directory, the next time wallets will be unlocked, which leads to inconsistent behaviour. To support the same initial conditions for all the tests I do `chrome_data` cleaning before each one. Please pay attention, that it doesn't apply to `plugins` directory, which is preserved between the tests — so basically I reset only the cache.

I suppose that in terms of the whole scope of UI tests this clean&untar procedure takes quite negligible time. Also if we parallel each test (parametrised) in future, then the wallet unlock step will be required before each one anyway.